### PR TITLE
force type text

### PR DESCRIPTION
**Pull request by vincentsarago**
possible fix for #59

### CODE REVIEW

File: tifeatures/layer.py [hunk 1]
```diff
@@ -165,21 +165,22 @@ def _where(
 
         # `ids` filter
         if ids is not None:
+            # Can't use "character varying" as cast type
+            # https://github.com/samuelcolvin/buildpg/issues/41
+            t = (
+                "text"
+                if self.id_column_info.type.startswith("character")
+                else self.id_column_info.type
+            )
```
Comment (vincentsarago) [inline tifeatures/layer.py:174]: Not 💯 sure this was needed here

**Comment by bitner** [comment]
Hey @vincentsarago this is also related to the PR I made to buildpg that never got merged https://github.com/samuelcolvin/buildpg/pull/38. Basically, buildpg does not handle any "identifier" (column name, table name, function name, type etc) that is not all lower case letters with no punctuation/spaces even when they are allowed (or as in the case of types like "character varying" even the canonical name).

I think in the long term, we do want to get proper identifier quoting upstream in buildpg, but the easier solution here would be to convert "character varying" to "text" at the time we are doing the sniffing.

We can do that in the big query in dbmodel.py by changing the type when it's sniffed. This hits every type that I can find that has a space in the name. If we do it here, we won't need to worry about it anywhere that we are using casts based on our sniffed types.

```
jsonb_build_object(
                                'name', attname,
                                'type', format_type(atttypid, null),
                                'description', col_description(attrelid, attnum)
                            )
```
to 
```
jsonb_build_object(
                                'name', attname,
                                'type', replace(replace(replace(replace(format_type(atttypid, null),'character varying','text'),'double precision','float8'),'timestamp with time zone','timestamptz'),'timestamp without time zone','timestamp'),
                                'description', col_description(attrelid, attnum)
                            )
```

**Comment by vincentsarago** [comment]
thanks @bitner 🙏 

I've updated the code with your proposed changes

File: tifeatures/dbmodel.py [hunk 2]
```diff
@@ -165,7 +165,7 @@ async def get_table_index(
                         jsonb_agg(
                             jsonb_build_object(
                                 'name', attname,
-                                'type', format_type(atttypid, null),
+                                'type', replace(replace(replace(replace(format_type(atttypid, null),'character varying','text'),'double precision','float8'),'timestamp with time zone','timestamptz'),'timestamp without time zone','timestamp'),
```
Comment (vincentsarago) [inline tifeatures/dbmodel.py:168]: I guess we need to update https://github.com/developmentseed/tifeatures/blob/master/tifeatures/dbmodel.py#L31 to add `float8`

**Review by bitner — APPROVED** [review]

**Pull request merged**